src/include/kdbiowrapper.h

```diff
@@ -0,0 +1,195 @@
+/**
+ * @file
+ *
+ * @brief
+ *
+ * @copyright BSD License (see LICENSE.md or https://www.libelektra.org)
+ */
+#ifndef KDB_IORAPPER_H_
+#define KDB_IORAPPER_H_
+
+struct ElektraIoWrapperFdInfo;
+
+/**
+ * Information about a file descriptor watched by IO-Wrapper
+ */
+typedef struct ElektraIoWrapperFdInfo
+{
+
+	/** Private data for library wrapper */
+	void * libraryWrapperData;
+
+	/** Private data for IO wrapper */
+	void * ioWrapperData;
+
+	/** File descriptor */
+	int fd;
+
+	/** Called when file descriptor state has changed */
+	void (*callback) (struct ElektraIoWrapperFdInfo * fdInfo, int flags);
+
+	/** Whether file descriptor is watched for changes, 0 means disabled */
+	int enabled;
+
+	/** Select on which file descriptor status changes the callback should be invoked */
+	int flags;
+} ElektraIoWrapperFdInfo;
+
+/**
+ * Available flags for file descriptors
+ */
+typedef enum {
+	ELEKTRA_IOWRAPPER_READABLE = 1 << 0,
+	ELEKTRA_IOWRAPPER_WRITABLE = 1 << 1,
+} ElektraWrapperFdFlags;
+
+struct ElektraIoWrapperTimeoutInfo;
+
+/**
+ * Information about a timeout in IO-Wrapper
+ */
+typedef struct ElektraIoWrapperTimeoutInfo
+{
+
+	/** Private data for library wrapper */
+	void * libraryWrapperData;
+
+	/** Private data for IO wrapper */
+	void * ioWrapperData;
+
+	/** Called when interval has elapsed */
+	void (*callback) (struct ElektraIoWrapperTimeoutInfo * timeoutInfo);
+
+	/** Whether timeout is enabled, 0 means disabled */
+	int enabled;
+
+	/**
+	 * Timeout interval in milliseconds (value > 0)
+	 */
+	unsigned int interval;
+
+} ElektraIoWrapperTimeoutInfo;
+
+struct ElektraIoWrapperIdleInfo;
+
+/**
+ * Information about an idle task in IO-Wrapper
+ */
+typedef struct ElektraIoWrapperIdleInfo
+{
+
+	/** Private data for library wrapper */
+	void * libraryWrapperData;
+
+	/** Private data for IO wrapper */
+	void * ioWrapperData;
+
+	/** Called when task can be performed */
+	void (*callback) (struct ElektraIoWrapperIdleInfo * idleInfo);
+
+	/** Whether timeout is enabled, 0 means disabled */
+	int enabled;
+
+} ElektraIoWrapperIdleInfo;
+
+struct ElektraIoWrapper;
+
+typedef struct ElektraIoWrapper
```

Comment:
In general very nice docu, but here explanation is missing.

Why do we need ElektraIoWrapper and its pointers to functions?